### DISCUSSION TITLE
BelongsToMany Searchable "SELECT *"

### DISCUSSION THREAD

**Discussion by henryavila** [laravel/nova-issues#3478] [Q&A] [answered]
Hi, I have a DB tha sotre some files on DB.
If I run a query with SELECT any field, except the binary fields, the query is super fast, but queries with SELECT * is slow because it will get the binary data content.

The problem is that When using 
```php
BelongsToMany::make('Name', 'relation', Resource::class)->searchable(),
```

It will generate a SQL with SELECT * FROM ...

How Can I setup BelongsToMany searchable to run a query withou SELECT * (in tis cau, I wann all fields, except the binary field)

**Comment by KasparRosin** [comment, marked as answer]
Would relatable query solve this?

If you put the relatableQuery function inside your `Resource::class`
```php
public static function relatableUsers(NovaRequest $request, $query)
{
    return $query->select('id', 'name');
}
```

**Reply by henryavila** [reply]
It workd perfectly. Thabs @KasparRosin!
